api/battle.php

```diff
@@ -13,80 +13,159 @@
+                $attack_id = $system->db->clean($_POST['target']);
+
+                // get user id off the attack link
+                $result = $system->db->query("SELECT `user_id` FROM `users` WHERE `attack_id`='{$attack_id}' LIMIT 1");
+                if ($system->db->last_num_rows == 0) {
+                    throw new RuntimeException("Invalid user!");
+                }
+
+                $attack_link = $system->db->fetch($result);
+                $attack_id = $attack_link['user_id'];
+
+                $user = User::loadFromId($system, $attack_id);
+                $user->loadData(User::UPDATE_NOTHING, true);
+
+                // check if the location forbids pvp
+                if ($player->current_location->location_id && $player->current_location->pvp_allowed == 0) {
+                    throw new RuntimeException("You cannot fight at this location!");
+                }
+
+                if ($user->village->name == $player->village->name) {
+                    throw new RuntimeException("You cannot attack people from your own village!");
+                }
+
+                if ($user->rank_num < 3) {
+                    throw new RuntimeException("You cannot attack people below Chuunin rank!");
+                }
+                if ($player->rank_num < 3) {
+                    throw new RuntimeException("You cannot attack people Chuunin rank and higher!");
+                }
+
+                if ($user->rank_num !== $player->rank_num) {
+                    throw new RuntimeException("You can only attack people of the same rank!");
+                }
+
+                if (!$user->location->equals($player->location)) {
+                    throw new RuntimeException("Target is not at your location!");
+                }
+                if ($user->battle_id) {
+                    throw new RuntimeException("Target is in battle!");
+                }
+                if ($user->last_active < time() - 120) {
+                    throw new RuntimeException("Target is inactive/offline!");
+                }
+                if ($player->last_death_ms > System::currentTimeMs() - (60 * 1000)) {
+                    throw new RuntimeException("You died within the last minute, please wait " .
+                        ceil((($player->last_death_ms + (60 * 1000)) - System::currentTimeMs()) / 1000) . " more seconds.");
+                }
+                if ($user->last_death_ms > System::currentTimeMs() - (60 * 1000)) {
+                    throw new RuntimeException("Target has died within the last minute, please wait " .
+                        ceil((($user->last_death_ms + (60 * 1000)) - System::currentTimeMs()) / 1000) . " more seconds.");
+                }
+
+                if ($system->USE_NEW_BATTLES) {
+                    BattleV2::start($system, $player, $user, Battle::TYPE_FIGHT);
+                } else {
+                    Battle::start($system, $player, $user, Battle::TYPE_FIGHT);
+                }
```

Comment:
We could put all this into a `TravelManager` method. 

The reason I think this particularly makes sense as a `Travel` type method is if we add functionality like variable attack distance, that's more related to the world and players moving around than it is the battle system itself. The battle system's concern starts and ends when a fight is kicked off (here on line 74-77)

Reply:
Moved to `TravelManager`.